dogen/cli.py

```diff
@@ -69,6 +70,19 @@ def run(self):
         parser.epilog = epilog
         args = parser.parse_args()
 
+        params = {}
+
+        if args.params:
+            for param in args.params:
+                if '=' not in param:
+                    self.log.error("The --param argument with value '%s' could not be parsed. Please make sure you use the 'KEY=value' format" % param)
+                    sys.exit(1)
+
+                k, v = param.split("=", 1)
+                params[k] = v
+
+        args.params = params
```

Comment:
Right. I guess the problem is that argparse doesn't support a dict directly. I could leave validation and just pass what I've got via CLI, but I guess I wanted to have the validation as close as possible to where argparse does it. I'm open for suggestions.